Tests/Functional/Action/Platform/OidcLoginAuthenticationActionTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Bundle\Lti1p3Bundle\Tests\Functional\Action\Platform;
+
+use Lcobucci\JWT\Parser;
+use OAT\Library\Lti1p3Core\Launch\Builder\OidcLaunchRequestBuilder;
+use OAT\Library\Lti1p3Core\Launch\Request\OidcLaunchRequest;
+use OAT\Library\Lti1p3Core\Link\ResourceLink\ResourceLink;
+use OAT\Library\Lti1p3Core\Message\LtiMessage;
+use OAT\Library\Lti1p3Core\Message\LtiMessageInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\Library\Lti1p3Core\Security\Jwt\AssociativeDecoder;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class OidcLoginAuthenticationActionTest extends WebTestCase
+{
+    /** @var KernelBrowser */
+    private $client;
+
+    /** @var RegistrationInterface */
+    private $registration;
+
+    protected function setUp(): void
+    {
+        $this->client = static::createClient();
+
+        $this->registration = static::$container
+            ->get(RegistrationRepositoryInterface::class)
+            ->find('testRegistration');
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->client->getCrawler()->clear();
+    }
+
+    public function testValidLoginAuthenticationWithPostMethod(): void
+    {
+        /** @var OidcLaunchRequest $oidcLaunchRequest */
+        $oidcLaunchRequest = static::$container->get(OidcLaunchRequestBuilder::class)->buildResourceLinkOidcLaunchRequest(
+            new ResourceLink('resourceLinkIdentifier'),
```

Comment:
If I'm not wrong this part could also be in a private because it's a duplicate between your Get and Post test.

But if you estimate that for clarity it's better to put it's ok also.

Reply:
in both case I depend on the `oidcLaunchRequest` to build the requests. I would let this next to them.

Reply:
ok for me